internal/provider/kubernetes/httproute.go

```diff
@@ -127,11 +127,14 @@ func (r *httpRouteReconciler) Reconcile(ctx context.Context, request reconcile.R
 	if errors.IsNotFound(err) {
 		log.V(2).Info("httproute not found, deleting it from the ResourceTable")
 		r.resources.HTTPRoutes.Delete(request.NamespacedName)
+		r.resources.Namespaces.Delete(request.Namespace)
 		return reconcile.Result{}, nil
 	}
 
 	log.V(2).Info("adding httproute to the ResourceTable")
-	r.resources.HTTPRoutes.Store(request.NamespacedName, httpRoute.DeepCopy())
+	copy := httpRoute.DeepCopy()
+	r.resources.HTTPRoutes.Store(request.NamespacedName, copy)
+	r.resources.Namespaces.Store(request.Namespace, namespaceFromName(copy.Namespace))
```

Comment:
this line seems to store the namespace from the httpRoute's namespace
Im trying to understand why the namespace is relevant in the translator API if the namespace and http route namespace will always match making this check 
https://github.com/envoyproxy/gateway/blob/9c34a8c5bd96e8dfb4fb34e0cea1995a951286f7/internal/gatewayapi/translator.go#L478
always compute to true (which is the only place Namespaces seems to be used today)

Reply:
The Namespace resources are needed because Gateway API gives Listeners the option to allow Routes to bind to them only if they're in a Namespace that matches a particular label selector (see https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.RouteNamespaces). Hence we need the Namespace, including its labels, to be able to check if it's allowed.

Because of that, I'm not sure that this approach of directly instantiating a Namespace will work - we really need to fetch it from the API server somehow, with its labels, in order for this part of the API to work.

Reply:
Possibly you want to do this in the HTTPRoute controller - list all the HTTPRoutes (as we just discussed on the call), then get each one's associated Namespace from the client/cache, then deliver the list of all Routes and all Namespaces to the resource translator. You could also do Services there, since we'll need those as well.

Reply:
> then get each one's associated Namespace from the client/cache

@skriss why do we need to get the httproute's associated ns object as opposed to building a list of local ns objects from the httproute's namespace? You mentioned something about checking ns labels, but I'm not following.